### ISSUE TITLE
Azure Kubernetes cluster gets replaced when upgrading from 1.19.11 to 1.20.9

### ISSUE THREAD

**Issue by arindamdat** [hashicorp/terraform-provider-azurerm#13438]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform - v0.13.6
azurerm - v2.76.0 

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Terraform Configuration Files

```hcl
terraform {
  required_version = "~> 0.13.0"
}
provider "null" {
  version = "~>2.1.2"
}

provider "random" {
  version = "~> 2.3.0"
  features {}
}

provider "local" {
  version = "~> 1.4"
}

provider "azurerm" {
  version = ">= 2.48.0"
  features {}
}
provider "azuread" {
  version = "~>0.7.0"
}
# RESOURCE AKS
resource "azurerm_kubernetes_cluster" "aks" {
  lifecycle {
    ignore_changes = [
      default_node_pool[0].node_count,
      windows_profile
    ]
  }

  name                            = "dev-euw-aks"
  location                        = "westeurope"
  resource_group_name             = "euw-rg"
  kubernetes_version              = "1.20.9"
  dns_prefix                      = "dev-euw-aks"
  tags                            = {
     Environment              = "Development"
  }

  linux_profile {
    admin_username = var.admin_user

    ssh_key {
      key_data = var.ssh_public_key
    }
  }

  # The windows_profile block should be optional.  However, there is a bug in the Terraform Azure provider
  # that does not treat this block as optional -- even if no windows nodes are used.  If not present, any
  # change that should result in an update to the cluster causes a replacement.
  windows_profile {
    admin_username = "azureuser"
    admin_password = var.windows_profile_admin_pwd
  }


  default_node_pool {
    name                  = "sysd4sv3001"
    orchestrator_version  = "1.20.9"
    node_count            = 1
    vm_size               = "Standard_D4S_v3"
    type                  = "VirtualMachineScaleSets"
    availability_zones    = ["1","2","3"]
    max_pods              = 250
    os_disk_size_gb       = 128
    vnet_subnet_id        = var.vnet_node_subnet_id
    node_labels           = var.default_node_pool.labels
    enable_auto_scaling   = var.default_node_pool.cluster_auto_scaling
    min_count             = var.default_node_pool.cluster_auto_scaling_min_count
    max_count             = var.default_node_pool.cluster_auto_scaling_max_count
    enable_node_public_ip = false
    upgrade_settings {
      max_surge = "33%"
    }
  }

  identity {
    type = "SystemAssigned"
  }

  role_based_access_control {
    enabled = true
  }

  addon_profile {
    oms_agent {
      enabled                    = false
    }
    kube_dashboard {
      enabled = false
    }
  }

  network_profile {
    load_balancer_sku  = "standard"
    outbound_type      = "loadBalancer"
    network_plugin     = "azure"
    network_policy     = "azure"
    dns_service_ip     = "10.201.0.10"
    docker_bridge_cidr = "172.201.0.1/16"
    service_cidr       = "10.201.0.0/16"
  }
}
```

### Debug Output

Debug contains secrets, so would prefer a private method if necessary for sharing

### Expected Behaviour

The kubernetes_version for the cluster and the orchestrator_version for the nodepool to be updated without recreating the cluster.
The expected plan should look like:
```hcl
# module.az-aks.azurerm_kubernetes_cluster.aks will be updated in-place
  ~ resource "azurerm_kubernetes_cluster" "aks" {
  ~ kubernetes_version              = "1.19.11" -> "1.20.9"
   ~ default_node_pool {
     ~ orchestrator_version         = "1.19.11" -> "1.20.9"
   }
}
```
### Actual Behaviour

The whole cluster gets replaced/recreated.
The actual plan looks like:
```hcl
# module.az-aks.azurerm_kubernetes_cluster.aks must be replaced
-/+ resource "azurerm_kubernetes_cluster" "aks" {
   - enable_pod_security_policy          = false -> null
   ~ fqdn                                = "dev-euw-aks-fa8e88f0.hcp.westeurope.azmk8s.io" -> (known after apply)
   ~ kubernetes_version                  = "1.19.11" -> "1.20.9"
   + private_dns_zone_id                 = (known after apply)
   + private_fqdn                        = (known after apply)
   ~ private_link_enabled                = false -> (known after apply)
    ~ default_node_pool {
	  - enable_host_encryption       = false -> null
          - fips_enabled                 = false -> null
	  + kubelet_disk_type            = (known after apply)
	  - node_taints                  = [] -> null
	  - only_critical_addons_enabled = false -> null
	  ~ orchestrator_version         = "1.19.11" -> "1.20.9"
         + os_sku                       = "Ubuntu" # forces replacement
	}
}
```
### Steps to Reproduce

1. `terraform apply`

### References

Seems related to the following Issue (not sure though)
* #13099

**Comment by arindamdat** [comment]
Hi @aristosvo , I guess you meant to say v2.77.0.  Even with azurerm - v2.77.0 I was getting this error. You can see below
![image](https://user-images.githubusercontent.com/7764767/134203743-596666d9-34b1-4d95-92f5-49196d6aeaa8.png)
![image](https://user-images.githubusercontent.com/7764767/134203931-9852bebc-560f-41f3-b2fc-56d378f26b00.png)
<br/>
After that, I explicitly added `os_sku` parameter to all nodepools and set it to `null`. Now, I get to see the expected Plan.<br/>
![image](https://user-images.githubusercontent.com/7764767/134204545-d5eae73c-2089-4572-8803-d21743288300.png)
![image](https://user-images.githubusercontent.com/7764767/134204701-68001101-b3fa-4b08-b9cb-7c2b38eb39d5.png)

**Comment by aristosvo** [comment]
I expect a `terraform refresh` to solve it, or (to have more control over it) use a newer version of `terraform` and roll `terraform plan/apply --refresh-only`

**Comment by arindamdat** [comment]
After rechecking, I confirm that this issue has been resolved with azurerm -v2.77.0. So, this ticket can be closed!
